From abf4c6f6d7f198a6c4502aa95cad0e1daa5c33fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 16 Dec 2020 21:45:44 +0100
Subject: [PATCH 1/4] Revert "theme: Add page padding to the icon grid"

This reverts commit e370adc76cc12eb4fd6f32afb1f87fab80791c4f.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541>
---
 data/theme/gnome-shell-sass/widgets/_app-grid.scss | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/data/theme/gnome-shell-sass/widgets/_app-grid.scss b/data/theme/gnome-shell-sass/widgets/_app-grid.scss
index c98980e0ec..c183cfeb09 100644
--- a/data/theme/gnome-shell-sass/widgets/_app-grid.scss
+++ b/data/theme/gnome-shell-sass/widgets/_app-grid.scss
@@ -8,10 +8,6 @@ $app_icon_size: 96px;
   column-spacing: $base_spacing * 6;
   max-row-spacing: $base_spacing * 12;
   max-column-spacing: $base_spacing * 12;
-  page-padding-top: $base_padding * 6;
-  page-padding-bottom: $base_padding * 6;
-  page-padding-left: $base_padding * 6;
-  page-padding-right: $base_padding * 6;
 }
 
 /* App Icons */
@@ -75,10 +71,6 @@ $app_grid_fg_color: #fff;
   & .icon-grid {
     row-spacing: $base_spacing * 2;
     column-spacing: $base_spacing * 5;
-    page-padding-top: 0;
-    page-padding-bottom: 0;
-    page-padding-left: 0;
-    page-padding-right: 0;
   }
 
   & .page-indicators {
-- 
GitLab


From f1ab9b74a4ea1d4323b7796f8cd915f3057d2dc2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 16 Dec 2020 21:45:52 +0100
Subject: [PATCH 2/4] Revert "iconGrid/iconGrid: Fill pages by default"

This reverts commit dd205d9f92eaf7197330f863142eb12df0ee3a40.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541>
---
 js/ui/iconGrid.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/js/ui/iconGrid.js b/js/ui/iconGrid.js
index ee70abee74..120a30d2c6 100644
--- a/js/ui/iconGrid.js
+++ b/js/ui/iconGrid.js
@@ -1329,9 +1329,9 @@ var IconGrid = GObject.registerClass({
             orientation: Clutter.Orientation.VERTICAL,
             columns_per_page: 6,
             rows_per_page: 4,
-            page_halign: Clutter.ActorAlign.FILL,
+            page_halign: Clutter.ActorAlign.CENTER,
             page_padding: new Clutter.Margin(),
-            page_valign: Clutter.ActorAlign.FILL,
+            page_valign: Clutter.ActorAlign.CENTER,
             last_row_align: Clutter.ActorAlign.START,
             column_spacing: 0,
             row_spacing: 0,
-- 
GitLab


From 7ef7ab9872f38f0fbed5faee35b5cb221576467d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 16 Dec 2020 21:46:05 +0100
Subject: [PATCH 3/4] Revert "iconGrid/iconGrid: Hook page-padding with CSS"

This reverts commit 541eaf59f22dcb8561cc7f5829883c848512c00c.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541>
---
 js/ui/iconGrid.js | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/js/ui/iconGrid.js b/js/ui/iconGrid.js
index 120a30d2c6..e9343046ea 100644
--- a/js/ui/iconGrid.js
+++ b/js/ui/iconGrid.js
@@ -1330,7 +1330,6 @@ var IconGrid = GObject.registerClass({
             columns_per_page: 6,
             rows_per_page: 4,
             page_halign: Clutter.ActorAlign.CENTER,
-            page_padding: new Clutter.Margin(),
             page_valign: Clutter.ActorAlign.CENTER,
             last_row_align: Clutter.ActorAlign.START,
             column_spacing: 0,
@@ -1462,12 +1461,6 @@ var IconGrid = GObject.registerClass({
 
         [found, value] = node.lookup_length('max-row-spacing', false);
         this.layout_manager.max_row_spacing = found ? value : -1;
-
-        const padding = new Clutter.Margin();
-        ['top', 'right', 'bottom', 'left'].forEach(side => {
-            padding[side] = node.get_length(`page-padding-${side}`);
-        });
-        this.layout_manager.page_padding = padding;
     }
 
     /**
-- 
GitLab


From f63812b0dc63e5c55120132ee099ed806cba3b1e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 16 Dec 2020 21:46:12 +0100
Subject: [PATCH 4/4] Revert "iconGrid/iconGridLayout: Add support for
 page-padding"

This reverts commit 72e2732bb695dda41ec138a99114d767e9543423.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1541>
---
 js/ui/iconGrid.js | 61 +++++++++++++----------------------------------
 1 file changed, 16 insertions(+), 45 deletions(-)

diff --git a/js/ui/iconGrid.js b/js/ui/iconGrid.js
index e9343046ea..da33eb6536 100644
--- a/js/ui/iconGrid.js
+++ b/js/ui/iconGrid.js
@@ -295,10 +295,6 @@ var IconGridLayout = GObject.registerClass({
             GObject.ParamFlags.READWRITE,
             Clutter.ActorAlign.$gtype,
             Clutter.ActorAlign.FILL),
-        'page-padding': GObject.ParamSpec.boxed('page-padding',
-            'Page padding', 'Page padding',
-            GObject.ParamFlags.READWRITE,
-            Clutter.Margin.$gtype),
         'page-valign': GObject.ParamSpec.enum('page-valign',
             'Vertical page align',
             'Vertical page align',
@@ -329,7 +325,6 @@ var IconGridLayout = GObject.registerClass({
             max_row_spacing: -1,
             orientation: Clutter.Orientation.VERTICAL,
             page_halign: Clutter.ActorAlign.FILL,
-            page_padding: new Clutter.Margin(),
             page_valign: Clutter.ActorAlign.FILL,
             row_spacing: 0,
             rows_per_page: 4,
@@ -344,7 +339,6 @@ var IconGridLayout = GObject.registerClass({
         this._maxRowSpacing = params.max_row_spacing;
         this._orientation = params.orientation;
         this._pageHAlign = params.page_halign;
-        this._pagePadding = params.page_padding;
         this._pageVAlign = params.page_valign;
         this._rowSpacing = params.row_spacing;
         this._rowsPerPage = params.rows_per_page;
@@ -421,11 +415,9 @@ var IconGridLayout = GObject.registerClass({
             }
 
             const emptyHSpace =
-                this._pageWidth - usedWidth - columnSpacingPerPage -
-                this._pagePadding.left - this._pagePadding.right;
+                this._pageWidth - usedWidth - columnSpacingPerPage;
             const emptyVSpace =
-                this._pageHeight - usedHeight -  rowSpacingPerPage -
-                this._pagePadding.top - this._pagePadding.bottom;
+                this._pageHeight - usedHeight -  rowSpacingPerPage;
 
             if (emptyHSpace >= 0 && emptyVSpace > 0)
                 return size;
@@ -594,30 +586,28 @@ var IconGridLayout = GObject.registerClass({
         const columnSpacingPerPage = this._columnSpacing * (nColumns - 1);
         const rowSpacingPerPage = this._rowSpacing * (nRows - 1);
 
-        const emptyHSpace =
-            this._pageWidth - usedWidth - columnSpacingPerPage -
-            this._pagePadding.left - this._pagePadding.right;
-        const emptyVSpace =
-            this._pageHeight - usedHeight -  rowSpacingPerPage -
-            this._pagePadding.top - this._pagePadding.bottom;
-        let leftEmptySpace = this._pagePadding.left;
-        let topEmptySpace = this._pagePadding.top;
+        let emptyHSpace = this._pageWidth - usedWidth - columnSpacingPerPage;
+        let emptyVSpace = this._pageHeight - usedHeight -  rowSpacingPerPage;
+        let leftEmptySpace;
+        let topEmptySpace;
         let hSpacing;
         let vSpacing;
 
         switch (this._pageHAlign) {
         case Clutter.ActorAlign.START:
+            leftEmptySpace = 0;
             hSpacing = this._columnSpacing;
             break;
         case Clutter.ActorAlign.CENTER:
-            leftEmptySpace += Math.floor(emptyHSpace / 2);
+            leftEmptySpace = Math.floor(emptyHSpace / 2);
             hSpacing = this._columnSpacing;
             break;
         case Clutter.ActorAlign.END:
-            leftEmptySpace += emptyHSpace;
+            leftEmptySpace = emptyHSpace;
             hSpacing = this._columnSpacing;
             break;
         case Clutter.ActorAlign.FILL:
+            leftEmptySpace = 0;
             hSpacing = this._columnSpacing + emptyHSpace / (nColumns - 1);
 
             // Maybe constraint horizontal spacing
@@ -626,7 +616,7 @@ var IconGridLayout = GObject.registerClass({
                     (this._maxColumnSpacing - this._columnSpacing) * (nColumns - 1);
 
                 hSpacing = this._maxColumnSpacing;
-                leftEmptySpace +=
+                leftEmptySpace =
                     Math.max((emptyHSpace - extraHSpacing) / 2, 0);
             }
             break;
@@ -634,17 +624,19 @@ var IconGridLayout = GObject.registerClass({
 
         switch (this._pageVAlign) {
         case Clutter.ActorAlign.START:
+            topEmptySpace = 0;
             vSpacing = this._rowSpacing;
             break;
         case Clutter.ActorAlign.CENTER:
-            topEmptySpace += Math.floor(emptyVSpace / 2);
+            topEmptySpace = Math.floor(emptyVSpace / 2);
             vSpacing = this._rowSpacing;
             break;
         case Clutter.ActorAlign.END:
-            topEmptySpace += emptyVSpace;
+            topEmptySpace = emptyVSpace;
             vSpacing = this._rowSpacing;
             break;
         case Clutter.ActorAlign.FILL:
+            topEmptySpace = 0;
             vSpacing = this._rowSpacing + emptyVSpace / (nRows - 1);
 
             // Maybe constraint vertical spacing
@@ -653,7 +645,7 @@ var IconGridLayout = GObject.registerClass({
                     (this._maxRowSpacing - this._rowSpacing) * (nRows - 1);
 
                 vSpacing = this._maxRowSpacing;
-                topEmptySpace +=
+                topEmptySpace =
                     Math.max((emptyVSpace - extraVSpacing) / 2, 0);
             }
 
@@ -1228,23 +1220,6 @@ var IconGridLayout = GObject.registerClass({
         this.notify('page-halign');
     }
 
-    // eslint-disable-next-line camelcase
-    get page_padding() {
-        return this._pagePadding;
-    }
-
-    // eslint-disable-next-line camelcase
-    set page_padding(padding) {
-        if (this._pagePadding.top === padding.top &&
-            this._pagePadding.right === padding.right &&
-            this._pagePadding.bottom === padding.bottom &&
-            this._pagePadding.left === padding.left)
-            return;
-
-        this._pagePadding = padding;
-        this.notify('page-padding');
-    }
-
     // eslint-disable-next-line camelcase
     get page_valign() {
         return this._pageVAlign;
@@ -1412,10 +1387,6 @@ var IconGrid = GObject.registerClass({
     }
 
     _findBestModeForSize(width, height) {
-        const { pagePadding } = this.layout_manager;
-        width -= pagePadding.left + pagePadding.right;
-        height -= pagePadding.top + pagePadding.bottom;
-
         const sizeRatio = width / height;
         let closestRatio = Infinity;
         let bestMode = -1;
-- 
GitLab

